google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -294,12 +294,16 @@ public ApiFuture<Void> closeAsync() {
     if (!isClosed()) {
       List<ApiFuture<Void>> futures = new ArrayList<>();
       if (isTransactionStarted()) {
-        futures.add(rollbackAsync());
+        try {
+          futures.add(rollbackAsync());
+        } catch (Exception exception) {
```

Comment:
What exception is specifically being thrown here by rollbackAsync(), in case rollback of transaction fails ? Does it make sense to catch that specific exception here ?

Reply:
It intentionally catches wide exceptions, as a number of things could happen:
1. The connection could be in a state where rolling back is not allowed (resulting in something like an `IllegalStateException`)
2. A backend error could happen (`SpannerException`)
3. The statement executor could reject the execution (`RejectedExecutionException`)

In addition, there could in theory be other unexpected exceptions, but we don't really care about them here, as it is more important to close the connection and clean up the resources that it is using, than surfacing an unexpected exception occurring in a method the user did not actively call (the user called `close()` and not `rollback()`).

Reply:
Sounds good!